### PR TITLE
Add gap to progressbar with multiple sections

### PR DESCRIPTION
**Pull request by lukasoppermann**
This bug was mentioned in [slack](https://github.slack.com/archives/CSGAVNZ19/p1726702236854379) and is apparent when looking at the [docs](https://primer.style/components/progress-bar).

**Expected**
When a progressbar has multiple sections, a 2px gap is shown between them.

**Current behavior**
When a progressbar has multiple sections, a **no gap** is shown between them. making them sometimes very hard to distinguish.

without gap

### Changelog
- added a 2opx gap when the Progressbar component has children.

| Before | After |
|--------|--------|
| before without gap | after with gap |

- added `sections` control to the playground story to allow users to check the difference.

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const ProgressContainer = styled.span<StyledProgressContainerProps>`
   background-color: ${get('colors.border.default')};
   border-radius: ${get('radii.1')};
   height: ${props => sizeMap[props.barSize || 'default']};
-
+  gap: '2px';
```
Comment (TylerJDev) [inline packages/react/src/ProgressBar/ProgressBar.tsx:55]:
```suggestion
  gap: 2px;
```

**Review by TylerJDev — APPROVED** [review]
Left a comment to fix the `gap` property not being applied, but otherwise looks good!

File: .changeset/stale-pets-tan.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react': patch
+---
+
+Adding default gap between sections for progressbars with more than one section
```
Comment (siddharthkp) [inline .changeset/stale-pets-tan.md:5]:
```suggestion
ProgressBar: Adding default gap between sections for progressbars with more than one section
```

**Review by siddharthkp — APPROVED** [review]
I see 2 suggestions, other than that. All good!

Approving in advance

**Pull request merged**